### ISSUE TITLE
[Feature Request] - Add cosine_distances_to_representatives to distances to different types of representative centroids for L2 normalized data

### ISSUE THREAD

**Issue by jolespin** [jolespin/skclust#25]
Example usage: 

```python
import numpy as np
import pandas as pd
from sklearn.preprocessing import normalize

# 6 observations in 3 groups, 4-dim embeddings
np.random.seed(42)
X_l2 = normalize(np.random.rand(6, 4), norm="l2")

labels = pd.Series(
    ["A", "A", "B", "B", "C", "C"],
    index=[f"obs_{i}" for i in range(6)],
    name="id_group",
)

# 2 representative types per group (e.g., core and union)
X_rep_l2 = normalize(np.random.rand(6, 4), norm="l2")

representative_labels = pd.Series(["A", "B", "C", "A", "B", "C"])
representative_types = pd.Series(["core", "core", "core", "union", "union", "union"])

df_distances = cosine_distances_to_representatives(
    X_l2=X_l2,
    labels=labels,
    X_representatives_l2=X_rep_l2,
    representative_labels=representative_labels,
    representative_types=representative_types,
)
print(df_distances)
#        core     union
# obs_0  ...      ...
# obs_1  ...      ...
# ...
```

**Comment by jolespin** [comment]
https://github.com/jolespin/skclust/pull/26
